library/aes.c

```diff
@@ -702,6 +720,47 @@ int mbedtls_aes_setkey_dec( mbedtls_aes_context *ctx, const unsigned char *key,
 
     return( ret );
 }
+
+#if defined(MBEDTLS_CIPHER_MODE_XTS)
+static int mbedtls_aes_xts_setkey( mbedtls_aes_xts_context *ctx,
+                                   const unsigned char *key,
+                                   unsigned int keybits,
+                                   int (*setkey1)( mbedtls_aes_context *ctx,
```

Comment:
I decided against that approach because the mode is already being checked earlier and it's not useful to check the mode again; it's better to get the most out of the check we did earlier. I also find this style to be more readable, at both the location where `mbedtls_aes_xts_setkey()` is called and where `setkey1()` is called.

Reply:
I've rewritten this to not use function pointers, to not check conditionals unecessarily, and yet still share a lot of code between encrypt and decrypt modes.